### PR TITLE
Add at::prod family of operations

### PR DESCRIPTION
**Pull request by vishwakftw**
I used #280 for reference.

### CODE REVIEW

File: torch_xla/csrc/ops/prod.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+namespace ir {
+namespace ops {
+namespace {
+
+xla::XlaOp LowerProd(const xla::XlaOp& input,
+                    const std::vector<xla::int64>& dimensions,
+                    bool keep_reduced_dimensions,
+                    c10::optional<at::ScalarType> dtype) {
+  xla::XlaOp casted_input;
+  casted_input = dtype ? ConvertTo(input, XlaHelpers::TypeOfXlaOp(input),
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/prod.cpp]: Like in sum.cpp, is a bit more readable. When you have ternaries which spans multiple lines, better and explicit "if".

File: torch_xla/csrc/aten_xla_type.cpp [hunk 2]
```diff
@@ -628,6 +628,44 @@ at::Tensor AtenXlaType::sum(const at::Tensor& self, at::IntArrayRef dim,
...
+      self_tensor,
+      xla::util::Iota<xla::int64>(self_tensor.shape().get().rank()),
+      /*keep_reduced_dimensions=*/false, c10::nullopt));
+}
+
+at::Tensor AtenXlaType::prod(const at::Tensor& self, int64_t dim,
+                             bool keepdim, at::ScalarType dtype) const {
+  return bridge::AtenFromXlaTensor(
+      XLATensor::prod(bridge::GetXlaTensor(self),
+                     xla::util::ToVector<xla::int64>(dim), keepdim, dtype));
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: When submitting code this way, please use clang-format with Google style.
This is what we use to format C++ code.

File: torch_xla/csrc/ops/prod.cpp (hunk 1 above)
Comment (vishwakftw) [reply torch_xla/csrc/ops/prod.cpp]: Thank you for the suggestion.

File: torch_xla/csrc/aten_xla_type.cpp (hunk 2 above)
Comment (vishwakftw) [reply torch_xla/csrc/aten_xla_type.cpp]: Thank you for the suggestion. I've made the necessary formatting changes, along with a few others which I encountered while formatting files in this diff. Hope that is fine.

**Pull request merged**